lib/controllers/messages.ts

```diff
@@ -197,6 +201,75 @@ type IGetMessagesHandler = (
   | IResponseErrorQuery
 >;
 
+type NotificationStatusHolder = {
+  readonly [channel in NotificationChannelEnum]?: NotificationChannelStatusValueEnum
+};
+
+/**
+ * Retrieve all notifications statuses (all channels) for a message.
+ *
+ * It makes one query to get the notification object associated
+ * to a message, then another query for each channel
+ * to retrieve the relative notification status.
+ *
+ * @returns an object with channels as keys and statuses as values
+ *          ie. { email: "SENT_TO_CHANNEL" }
+ */
+async function getMessageNotificationStatuses(
+  notificationModel: NotificationModel,
+  notificationStatusModel: NotificationStatusModel,
+  retrievedMessage: RetrievedMessage
+): Promise<Either<Error, Promise<NotificationStatusHolder>>> {
+  const errorOrMaybeNotification = await notificationModel.findNotificationForMessage(
+    retrievedMessage.id
+  );
+  if (isRight(errorOrMaybeNotification)) {
+    // It may happen that the notification object is not yet created in the database
+    // due to some latency, so it's better to not fail here but return an empty object
+    const maybeNotification = errorOrMaybeNotification.value;
+    if (isNone(maybeNotification)) {
+      winston.info(
+        `getMessageNotificationStatuses|Notification not found|messageId=${
+          retrievedMessage.id
+        }`
+      );
+      return right(Promise.resolve({}));
```

Comment:
I think it makes sense to have an `Option` here to reflect the semantic of `maybeNotification` (that is an `Option` as well), consider having this method return a `Promise<Either<Error, Option<NotificationStatusHolder>>>`